content/chainguard/chainguard-images/how-to-use/use-with-openshift.md

```diff
@@ -0,0 +1,78 @@
+---
+title: "How to Use Chainguard Containers with OpenShift"
+linktitle: "Using With OpenShift"
+type: "article"
+description: "Special considerations for using Chainguard Containers with OpenShift"
+date: 2025-06-17T08:49:31+00:00
+lastmod: 2025-06-17T08:49:31+00:00
+draft: false
+tags: ["Chainguard Containers", "OpenShift", "Product"]
+images: []
+menu:
+  docs:
+    parent: "chainguard-images"
+weight: 008
+toc: true
+---
+
+When [Using Chainguard Containers](./how-to-use-chainguard-images.md) with OpenShift, there are some adjustments that need to be made to the usual process.
+
+# Adjust Ownership and Permissions
+
+By default, OpenShift Container Platform runs containers using an arbitrarily assigned User ID (UID), as described in the [Red Hat documentation](https://docs.redhat.com/en/documentation/openshift_container_platform/4.17/html/images/creating-images#use-uid_create-images).
+
+There are required access settings for an image to support running as an arbitrary user:
+
+- Directories and files that are written to by processes in the image must be owned by the `root` group and that group must have both `read` and `write` permissions
+- Files that will be executed must also have group execute permissions
+
+Following the Red Hat requirements, to use Chainguard Containers you must make the following change in your Dockerfile to set the required ownership and permissions:
+
+```
+RUN chgrp -R 0 /some/directory && \
+    chmod -R g=u /some/directory
+```
+
+# Create `/app` Directory and `HOME`
+
+When running on OpenShift clusters, you will find that the OpenShift user cannot create config files inside their home directory. This is because [OpenShift is designed to start container instances using a random User ID](https://www.redhat.com/en/blog/a-guide-to-openshift-and-uids).
+
+To avoid this being an issue when using Chainguard Containers:
+
+1. Set a `HOME` variable for the user (it would otherwise be set as `/` for root)
+1. Create an `/app` directory in every Dockerfile
+1. Set `/app` directory permissions to `755` and ownership to `65532:0`
+
+This can help you avoid or limit switching to the `root` user during the build phase when no package installation is required.
+
+ Here's a sample Dockerfile excerpt covering this process.
+
+ ```
```

Comment:
Can you also have a pretty small complete Dockerfile that would include the full context?

Reply:
That could be after the snippet, I would just add some prose to explain

Reply:
@anitapower , I got the snippet in question from you. Any chance you have a full Dockerfile with context? I can sanitize it and abbreviate it to "small complete" status.